lib/view_component/collection.rb

```diff
@@ -11,18 +11,22 @@ def render_in(view_context, &block)
       @component.validate_collection_parameter!(validate_default: true)
 
       @collection.map do |item|
+        if !iterator.first? && @spacer_component
+          spacer_content = @spacer_component.new(item: item, index: iterator.index).render_in(view_context, &block)
```

Comment:
Will all spacer components need to accept an `item` parameter? IMO we should look at using `component_options` or something like it here.

Reply:
Is there a way to do this without making parameters required? The component_options is close to what I'd want, but I don't want the addition of parameters to break existing component usages.

Reply:
What do you mean by:

> I don't want the addition of parameters to break existing component usages

`component_options` should only pass in the options if they are present in the component method signature.

Reply:
Sorry it took me so long to loop back on this -- Basically, my preference would be that there is a set of ambient info that a component could use -- maybe a given spacer doesn't care about index or item, for example, so it wouldn't need a parameter for those. I guess i don't understand how component options helps :D